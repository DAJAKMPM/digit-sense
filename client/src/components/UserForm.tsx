import React from "react";
import { Box, TextField, Button } from "@mui/material";
import { useNavigate } from "react-router-dom";
import { UserFormProps } from "../types";

const UserForm = ({
  mode,
  user,
  handleSubmit,
  handleChange,
}: UserFormProps) => {
  const navigate = useNavigate();

  const handleCancel = () => navigate("/");

  return (
    <Box component="form" onSubmit={handleSubmit} noValidate sx={{ mt: 1 }}>
      <TextField
        margin="normal"
        required
        fullWidth
        id="name"
        label="Full Name"
        name="name"
        onChange={handleChange("name")}
        autoComplete="name"
        value={user?.name || ""}
        autoFocus
      />
      <TextField
        margin="normal"
        required
        fullWidth
        name="email"
        label="Email"
        type="email"
        id="email"
        onChange={handleChange("email")}
        value={user?.email || ""}
        autoComplete="email"
      />
      <TextField
        margin="normal"
        required
        fullWidth
        name="address"
        label="Address"
        id="address"
        onChange={handleChange("address")}
        value={user?.address || ""}
        autoComplete="address"
      />
      <TextField
        margin="normal"
        required
        fullWidth
        name="age"
        label="Age"
        type="number"
        id="age"
        onChange={handleChange("age")}
        value={user?.age || 0}
        autoComplete="age"
      />

      <Button type="submit" fullWidth variant="contained" sx={{ mt: 3, mb: 2 }}>
        {mode === "add" ? "Save" : "Update"}
      </Button>
      <Button
        type="button"
        fullWidth
        variant="contained"
        color="inherit"
        onClick={handleCancel}
      >
        Cancel
      </Button>
    </Box>
  );
};

export { UserForm };
