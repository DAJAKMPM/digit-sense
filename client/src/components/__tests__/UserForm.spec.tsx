import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import { UserForm } from "../UserForm";

test("should match the UserForm snapshot properly", () => {
  const { container } = render(
    <Router>
      <UserForm
        mode="add"
        user={null}
        handleChange={() => () => {}}
        handleSubmit={() => {}}
      />
    </Router>
  );
  expect(container).toMatchSnapshot();
});
