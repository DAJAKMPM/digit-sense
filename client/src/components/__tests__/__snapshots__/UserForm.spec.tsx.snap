// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the UserForm snapshot properly 1`] = `
<div>
  <form
    class="MuiBox-root css-164r41r"
    novalidate=""
  >
    <div
      class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-focused Mui-required css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="true"
        for="name"
        id="name-label"
      >
        Full Name
        <span
          aria-hidden="true"
          class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth Mui-focused MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          autocomplete="name"
          class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          id="name"
          name="name"
          required=""
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-1in441m"
          >
            <span>
              Full Name
               
              *
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        for="email"
        id="email-label"
      >
        Email
        <span
          aria-hidden="true"
          class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          autocomplete="email"
          class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          id="email"
          name="email"
          required=""
          type="email"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-1ftyaf0"
          >
            <span>
              Email
               
              *
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        for="address"
        id="address-label"
      >
        Address
        <span
          aria-hidden="true"
          class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          autocomplete="address"
          class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          id="address"
          name="address"
          required=""
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-1ftyaf0"
          >
            <span>
              Address
               
              *
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled Mui-required css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="true"
        for="age"
        id="age-label"
      >
        Age
        <span
          aria-hidden="true"
          class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          autocomplete="age"
          class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          id="age"
          name="age"
          required=""
          type="number"
          value="0"
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-1in441m"
          >
            <span>
              Age
               
              *
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth css-1vhaqj4-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="submit"
    >
      Save
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedInherit MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorInherit MuiButton-fullWidth css-1nalwf2-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Cancel
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </form>
</div>
`;
